### ISSUE TITLE
about sequence labeling

### ISSUE THREAD

**Issue by xiayandi** [Fematich/nn_ner#2]
Hi Fematich,

Your code is amazing, and I happen to be doing a project using deep learning with theano. 

There is one question. It seems that you implemented a "if .... else ..." version of sequence labeling for "BIOES" encoding in generate_output() function. How do you deal with the situation if 'O' followed by 'I', which is a wrong case?

And have you tried the viterbi fashion sequence labeling like adding a transition matrix and train it during gradient descent as described in the paper "Natural language processing almost from scratch"?

Thank you.

**Comment by Fematich** [comment]
Hi Xiayandi,

Thanks!

The generate_output() function just transforms the IOBES encoding generated by the Theano-code to the IOB-format that is used with the CoNLL dataset(http://www.cnts.ua.ac.be/conll2003/ner/), as such, I do not do any validation checks on the output of the Theano-code.

I have not tried the Viterbi sequence labeling unfortunately. At the moment I don't have time to do this, but if you are willing to share this, that would be great!

If you have any further questions, I am happy to answer them...

Kind regards,
Matthias
